dspace-server-webapp/src/main/java/org/dspace/app/rest/ProcessRestController.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.ProcessRest;
+import org.dspace.app.rest.model.hateoas.ProcessResource;
+import org.dspace.app.rest.repository.ProcessRestRepository;
+import org.dspace.authorize.AuthorizeException;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller takes cares of all the request going to the Processes endpoint
+ */
+@RestController
+@RequestMapping("/api/" + ProcessRest.CATEGORY + "/" + ProcessRest.PLURAL_NAME)
+public class ProcessRestController implements InitializingBean {
```

Comment:
I'm confused why we need a custom Controller here.  Shouldn't we be able to simply use `ProcessRestRepository` and `ProcessRest` for all the various `GET` requests?  See for example `ItemRestRepository`  and `ItemRest` (and not that we do NOT have a separate `ItemRestController`).

Reply:
In the current code base we would indeed be able to remove this ProcessRestController and implement the current methods (which simply map to a findAll and findOne) through the ProcessRestRepository and ProcessRest/ProcessResource similar to the Item endpoint equivalents by utilising the generic RestResourceController. However in the future we'll need to expose bitstreams of the process, outputs of the process, filter the process' bitstreams by type and return these, return a specific bitstream found by name for the given process etc etc. These will all have certain authorization criteria and the links need to be hidden if the current user is not allowed to view this information. Once we implement this, we'll need to move it to a custom controller since the logic becomes a lot more complex and fine-tuned over time when we implement all those use cases.
Currently for the logic that is available within the ProcessRestController, we could indeed move it into this framework; however we opted not to do this yet when we were looking at the future. However we can still move it into the framework and pull it apart later on if this is desired.
As for the ScriptRestController specifically, this is not possible as the method within this endpoint would translate to a POST on a rel which is not supported yet within the framework.